### ISSUE TITLE
cmd/cgo: ignores arguments in $CC

### ISSUE THREAD

**Issue by hirochachacha** [golang/go#14551]
Please answer these questions before submitting your issue. Thanks!
1. What version of Go are you using (`go version`)?
   go version devel +ff27421 Sun Feb 28 11:06:40 2016 +0000 darwin/amd64
2. What operating system and processor architecture are you using (`go env`)?
   GOARCH="amd64"
   GOBIN=""
   GOEXE=""
   GOHOSTARCH="amd64"
   GOHOSTOS="darwin"
   GOOS="darwin"
   GOPATH="/Users/hiro/.go"
   GORACE=""
   GOROOT="/Users/hiro/work/go"
   GOTOOLDIR="/Users/hiro/work/go/pkg/tool/darwin_amd64"
   CC="clang"
   GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -gno-record-gcc-switches -fno-common"
   CXX="clang++"
   CGO_ENABLED="1"
3. What did you do?
   (Use play.golang.org to provide a runnable example, if possible.)

`$ CC="$(xcrun --sdk iphoneos --find clang) -arch armv7 -isysroot $(xcrun --sdk iphoneos --show-sdk-path)" GOARCH=arm go tool cgo -godefs input.go`

input.go

```
package p

/*
typedef void * ptr;
*/
import "C"

type Ptr C.ptr
```
1. What did you expect to see?
   No errors.
2. What did you see instead?

```
unexpected: 8-byte pointer type - *void
```

XCode 7.2.1 (latest)

```
$(xcrun --sdk iphoneos --find clang) --version

Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin15.3.0
Thread model: posix
```

Thank you.

**Comment by ianlancetaylor** [comment]
Your C compiler seems to be generating debug info that says that the size of the type "void*" is 8 bytes.  cgo is complaining because you set GOARCH=arm, so cgo expects 4 byte pointers.

Look closely at the compiler in

    CC="$(xcrun --sdk iphoneos --find clang) -arch armv7 -isysroot $(xcrun --sdk iphoneos --show-sdk-path)"

Are you sure that compiler will generate 32-bit ARM code?  (I don't know who XCode works myself).

You may want to run cgo with `-debug-gcc` to see precisely how cgo is invoking the compiler.

**Comment by hirochachacha** [comment]
>You may want to run cgo with -debug-gcc to see precisely how cgo is invoking the compiler.

It helps me.

I guess I understand the problem.

$ `CC="foo bar" go env CC`
```
foo
```

Is this work as intended?

**Comment by hirochachacha** [comment]
`CXX` has same problem.

**Comment by minux** [comment]
You need to use misc/ios/clangwrap.sh as CC for darwin/arm
and darwin/arm64.

**Comment by hirochachacha** [comment]
Yes. But is this desirable behavior?

**Comment by ianlancetaylor** [comment]
No, it's not.  But the cgo tool does provide another way to specify compiler options, so it's not urgent.

**Comment by seankhliao** [comment]
resolved with #43808 and [CL 284780](https://golang.org/cl/284780)
